src/libraries/System.Net.Http/ref/System.Net.Http.cs

```diff
@@ -901,3 +905,19 @@ public partial class WarningHeaderValue : System.ICloneable
         public static bool TryParse([System.Diagnostics.CodeAnalysis.NotNullWhenAttribute(true)] string? input, [System.Diagnostics.CodeAnalysis.NotNullWhenAttribute(true)] out System.Net.Http.Headers.WarningHeaderValue? parsedValue) { throw null; }
     }
 }
+namespace System.Net.Http.Metrics
+{
+    public class HttpMetricsEnrichmentContext
+    {
+        public HttpRequestMessage Request { get { throw null; } }
+
+        public HttpResponseMessage? Response { get { throw null; } }
+
+        public Exception? Exception { get { throw null; } }
+        public ICollection<KeyValuePair<string, object?>> CustomTags { get { throw null; } }
```

Comment:
Is this the best way to expose this?  Are there scenarios for folks changing/removing/enumerating tags previously added, or only for adding additional ones?

Reply:
> changing/removing

Those scenarios will be addressed by a general-purpose metrics renaming feature @noahfalk is working on. We agreed to not mix this concern into the HttpClient enrichment feature.

> enumerating

All the information standard tags are based are based on is available in the request and the response. If other information would be needed in the future, we can add it to the context in form of new properties.

Reply:
That's not where I was going with my question.  We're exposing this as ` ICollection<KeyValuePair<string, object?>>`.  Is that needed, or would a `AddCustomTag(string key, object value)` be sufficient?

Reply:
@noahfalk @JamesNK thoughts?

Reply:
The only concrete scenario I am aware of is R9, and their needs are strictly to add new tags, not enumerating/changing/removing existing tags. I think its fine if we had AddCustomTag() instead of ICollection.

Reply:
I think just adding `AddCustomTag(string key, object value)` would be enough. While there is a feature in dotnet/extensions enrichment to validate tags aren't duplicated, the validation is done when resolving the enrichers from DI, not per-request: https://github.com/dotnet/extensions/blob/99d8ccd0b8a0154c2f0336f5ec2e4c0a1d8729da/src/Libraries/Microsoft.Extensions.Http.Telemetry/Metering/HttpMeteringHandler.cs#L94-L103. It doesn't need read access from the tags to work.

If a new scenario comes up in the future where getting/settings/removing/enumerating is needed, how would the API evolve? If required, it looks like there is nothing stopping the context from adding a collection in the future, and `AddCustomTag` continues to add to it.

Reply:
Updated #86281 to expose `AddCustomTag(string key, object? value)` instead. If later it turns out that we need to enumerate (which I feel is unlikely based on the discussion), we can expose a collection and potentially obsolete the single-tag method.